doc/level0.rst

```diff
@@ -238,8 +238,34 @@ Autocrypt-compatible agents SHOULD track and store in
 necessarily the literal header emitted (for the literal header, see
 next section).  The ``pah`` MUST contain the following fields:
 
- * ``key`` -- the raw key material, after base64 decoding
- * ``prefer_encrypted`` -- a tri-state: ``nopreference``, ``yes``, or ``no``
+ * ``key``: the raw key material, after base64 decoding
+ * ``prefer_encrypted``: a quad-state: ``nopreference``, ``yes``, ``no`` or ``reset``
+
+Optionally, an agent MAY store and maintain the following data in
+order to provide clearer feedback to the user:
+
+ * ``counting_since``: The UTC timestamp of when we started counting
+ * ``count_have_ach``: A count of parsed AutoCrypt headers
+ * ``count_no_ach``: A count of messages without AutoCrypt headers
+ * ``bad_user_agent``: The apparent user-agent (if known) of the last
```

Comment:
How is a mail user agent expected to accumulate these counts, given that it might see the same message multiple times?  From the algorithms described here, i don't see a way to avoid double-counting when the MUA happens to look at a message twice.